readme update (#1)
hgraph-io/hedera-walletconnect-utils: README.md

> This package is a wrapper library for the [`@hashgraph/sdk`](https://www.npmjs.com/package/@hashgraph/sdk) library. It provides a means of integrating Hedera with WalletConnect with functions to facilitate implementing the [Hedera JSON-RPC schema](https://specs.walletconnect.com/2.0/blockchain-rpc/everscale-rpc). To see how to set up WalletConnect in dApps/wallets, view their [documentation](https://docs.walletconnect.com/2.0/). Once the dApp and wallet are configured to manage WalletConnect sessions, you can use this module's functions to easily create and handle sessions requests for Hedera.

Reviewer: ```suggestion
This package is a wrapper library for the [`@hashgraph/sdk`](https://www.npmjs.com/package/@hashgraph/sdk) library. It provides a means of integrating Hedera and WalletConnect with functions to facilitate implementing the [Hedera JSON-RPC schema](https://specs.walletconnect.com/2.0/blockchain-rpc/everscale-rpc). To see how to set up WalletConnect in dApps/wallets, view their [documentation](https://docs.walletconnect.com/2.0/). Once the dApp and wallet are configured to manage WalletConnect sessions, you can use this module's functions to easily create and handle sessions requests for Hedera.
```

## DApp Usage

1. **Initialization of WalletConnect SignClient**: Start by setting up a WalletConnect [`SignClient`](https://docs.walletconnect.com/2.0/api/sign/dapp-usage). This is your primary interface for establishing and managing sessions between your dApp and the user's wallet.

```js
import SignClient from '@walletconnect/sign-client'

const signClient = await SignClient.init({ ...signClientProps })
// Ensure other initialization steps are followed as per the WalletConnect documentation.
```

> 2. **Constructing a Hedera Transaction**: Use the `@hashgraph/sdk` to build your desired Hedera transaction. It's important to manually set a `TransactionId` as it ensures traceability and potential reconciliation of transactions. [See the SDK documentation for more transaction types](https://docs.hedera.com/hedera/sdks-and-apis/sdks/transactions).

Reviewer: I think setting a `TransactionId` is required when not using a signer (which we won't have)?

  .setTransactionId(transactionId)
```

> 3. **Building the Session Request Payload**: The `@hgraph.io/hedera-walletconnect-utils` library provides a seamless way to prepare the session request payload. Ensure that you set the `RequestType` accurately to match the type of Hedera transaction you've constructed.

Reviewer: change to `@hashgraph/wallectconnect` as we'll be moving to their repo

> 3. **Building the Session Request Payload**: The `@hgraph.io/hedera-walletconnect-utils` library provides a seamless way to prepare the session request payload. Ensure that you set the `RequestType` accurately to match the type of Hedera transaction you've constructed.

Reviewer: ```suggestion
3. **Building the Session Request Payload**: The `@hashgraph/walletconnect` library provides a seamless way to prepare the session request payload. Ensure that you set the `RequestType` accurately to match the type of Hedera transaction you've constructed.
```

})
```

By following these steps, developers can ensure a comprehensive and user-friendly integration of Hedera with WalletConnect in their dApps. Always refer to the linked documentation for in-depth details and best practices.

> 

Reviewer: nice

## Wallet Usage

### 1. Setup and Installation

First, make sure you've installed the necessary npm packages:

```bash

> npm install @walletconnect/sign-client @hashgraph/sdk @hgraph.io/hedera-walletconnect-utils

Reviewer: update to hedera repo

> npm install @walletconnect/sign-client @hashgraph/sdk @hgraph.io/hedera-walletconnect-utils

Reviewer: ```suggestion
npm install @walletconnect/sign-client @hashgraph/sdk @hashgraph/walletconnect
```

```

> If you have WalletConnect v1.x installed, consider following the [migration guide](https://docs.walletconnect.com/2.0/api/auth/wallet-usage) for a seamless transition to v2.x.

Reviewer: ```suggestion
```

> 

Reviewer: this is for hedera specifically, and no wallet shoudl have wallet connect v1 integrated, so I think it makes sense to leave this out

```javascript
import SignClient from '@walletconnect/sign-client';

const signClient = await SignClient.init({
  projectId: 'YOUR_PROJECT_ID',
  metadata: {
    name: 'Your Wallet Name',
    description: 'Description for your wallet',
    url: 'https://your-wallet-url.com',
    icons: ['https://your-wallet-url.com/icon.png'],
  }
});
```

### 3. Event Listeners

WalletConnect emits various events during a session. Listen to these events to synchronize the state of your application:

```javascript
// Handle pairing proposals
signClient.on('session_proposal', event => {
  // Display session proposal to the user and decide to approve or reject
});

// Handle session requests, like signing transactions or messages
signClient.on('session_request', event => {
  // Process the session request
});

// Handle session deletions
signClient.on('session_delete', event => {
  // React to session termination
});
```

For a complete list of events and their structure, refer to the provided WalletConnect tutorial. [WalletConnect Usage](https://docs.walletconnect.com/2.0/api/auth/wallet-usage)

### 4. Pairing with dApps

> Pairing establishes a connection between the wallet and a dapp. Once paired, the dapp can send session requests to the wallet.

Reviewer: dapp -> dApp

> Pairing establishes a connection between the wallet and a dapp. Once paired, the dapp can send session requests to the wallet.

Reviewer: ```suggestion
Pairing establishes a connection between the wallet and a dApp. Once paired, the dApp can send session requests to the wallet.
```

#### a. Pairing via URI

> If a dapp shares a URI for pairing:

Reviewer: dapp -> dApp

> If a dapp shares a URI for pairing:

Reviewer: ```suggestion
If a dApp shares a URI for pairing:
```

```javascript
await signClient.core.pairing.pair({ uri: 'RECEIVED_URI' });
```

Upon successful pairing, the `session_proposal` event will be triggered.

#### b. Pairing via QR Codes

For a better user experience, dApps often share QR codes that wallets can scan to establish a pairing. Use a QR code scanning library to scan and obtain the URI, then proceed with pairing:

```javascript
const scannedUri = '...'; // URI obtained from scanning the QR code
await signClient.core.pairing.pair({ uri: scannedUri });
```

### 5. Handling Session Proposals

Upon receiving a `session_proposal` event, display the proposal details to the user. Allow them to approve or reject the session:

```javascript
// Approving a session proposal
const { topic, acknowledged } = await signClient.approve({
  id: proposalId, // From the session_proposal event
  namespaces: {
    eip155: {
      accounts: ['eip155:1:0xYOUR_HEDERA_ACCOUNT_ID'],
      methods: ['personal_sign', 'eth_sendTransaction'],
      events: ['accountsChanged']

>     }

Reviewer: update to hedera

  }
});

// Rejecting a session proposal
await signClient.reject({
  id: proposalId,
  reason: {
    code: 1,
    message: 'User rejected the proposal'
  }
});
```

### 6. Handling Session Requests

Upon receiving a `session_request` event, process the request. For instance, if the dApp requests a transaction to be signed:

```javascript
// Using the @hgraph.io/hedera-walletconnect-utils library

> import { base64StringToTransaction, HederaWallet } from '@hgraph.io/hedera-walletconnect-utils';

Reviewer: `@hashgraph/walletconnect`

> import { base64StringToTransaction, HederaWallet } from '@hgraph.io/hedera-walletconnect-utils';

Reviewer: ```suggestion
import { base64StringToTransaction, HederaWallet } from '@hashgraph/walletconnect';
```
